src/cdomains/apron/linearTwoVarEqualityDomain.apron.ml

```diff
@@ -0,0 +1,952 @@
+                 else subst_var ts h1 (Some h2, Z.(b + (b2 - b1))))))
+    in 
+    match t1.d, t2.d with
+    | None, _ -> { d = None; env = sup_env} 
+    | _, None -> { d = None; env = sup_env} 
+    | Some d1', Some d2' -> 
+      let ds = ref (Some (Array.copy d1')) in
+      if Array.length d2' <> 0 then
+        for j = 0 to Array.length d2' - 1 do
+          add_conj ds d2'.(j) j
+        done; 
+      {d = !ds; env = sup_env} 
+
+  let meet t1 t2 =
+    let res = meet t1 t2 in
+    if M.tracing then M.tracel "meet" "meet a: %s b: %s -> %s \n" (show t1) (show t2) (show res) ;
+    res
+
+  let meet t1 t2 = timing_wrap "meet" (meet t1) t2
+
+  let leq t1 t2 =
+    let env_comp = Environment.compare t1.env t2.env in (* Apron's Environment.compare has defined return values. *)
+    let implies ts t i : bool =
+      match t with
+      | (None, b) -> 
+        (match ts.(i) with
+         | (None, b') -> Z.equal b b'
+         | _ -> false)
+      | (Some j, b) -> 
+        (match ts.(i), ts.(j) with
+         | (None, b1), (None, b2) -> Z.equal b1 (Z.add b2 b)
+         | (Some h1, b1), (Some h2, b2) ->
+           h1 = h2 && Z.equal b1 (Z.add b2 b)
+         | (Some _, _), (_, _) -> false
+         | (_, _), (Some _, _) -> false
+        )
+    in  
+    if env_comp = -2 || env_comp > 0 then false else
+    if is_bot_env t1 || is_top t2 then true else
+    if is_bot_env t2 || is_top t1 then false else (
+      let m1, m2 = Option.get t1.d, Option.get t2.d in
+      let m1' = if env_comp = 0 then m1 else dim_add (Environment.dimchange t1.env t2.env) m1 in
+      let result : bool ref = ref true in
+      for i = 0 to Array.length m2 - 1 do
+        let t = m2.(i) in
+        if not (implies m1' t i) then result := false;
+      done;
+      !result)
+
+  let leq a b = timing_wrap "leq" (leq a) b
+
+  let leq t1 t2 =
+    let res = leq t1 t2 in
+    if M.tracing then M.tracel "leq" "leq a: %s b: %s -> %b \n" (show t1) (show t2) res ;
+    res
+
+  let join a b = 
+    let ts_zip t1 t2 =
+      if Array.length t1 <> Array.length t2 then None else
+        let zts = Array.init (Array.length t1) (fun (i : int) -> (i, t1.(i), t2.(i))) in
+        Some zts
+    in
+    let const_offset t = match t with
+      | (_, b) -> b 
+    in
+    let diff t1 t2 = Z.((const_offset t1) - (const_offset t2))
+    in
+    let cmp_z x y = 
+      let cmp_z_ref x y: int =
+        match x, y with
+        | (None, _), (None, _) -> 0
+        | (None, _), (Some _, _) -> -1
+        | (Some _, _), (None, _) -> 1
+        | (Some ii, _), (Some ij, _) -> ii - ij 
+      in
+      match x, y with
+      | (_, t1i, t2i), (_, t1j, t2j) -> 
+        let diff_e1 = cmp_z_ref t1i t1j in
+        if diff_e1 <> 0 then diff_e1 else
+          let diff_e2 = cmp_z_ref t2i t2j in
+          if diff_e2 <> 0 then diff_e2 else 
+            Z.to_int (Z.((diff t1i t2i) - (diff t1j t2j)))
+    in
+    let sort_z_by_expr zts =
+      match zts with
```

Comment:
You may use `Option.iter` instead of doing this match statement.

Reply:
that should be possible but somehow the function is not found during compilation. I have already run opam update and upgrade and completely rebuild the project but no difference.

Reply:
`Option.iter` is part of the standard library since OCaml 4.08, and we are using 4.14, so it should definitely be there. Did you check for typos?

Reply:
If there's `open Batteries`, then it's not available since `BatOption` does not have `iter` for some reason, but just `may`. So `BatOption.may` or `Stdlib.Option.iter` should work.

Reply:
Works :)